packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-endpoint-config.ts

```diff
@@ -0,0 +1,135 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import { ProductionVariant } from './base-types';
+
+/**
+ * Properties for creating an Amazon SageMaker endpoint configuration
+ *
+ * @experimental
+ */
+export interface SageMakerCreateEndpointConfigProps extends sfn.TaskStateBaseProps {
+  /**
+   * The name of the endpoint configuration.
+   */
+  readonly endpointConfigName: string;
+  /**
+   * The Amazon Resource Name (ARN) of a AWS Key Management Service key that Amazon SageMaker
+   * uses to encrypt data on the storage volume attached to the ML compute instance that hosts the endpoint.
+   *
+   * @default - None
+   */
+  readonly kmsKeyId?: string;
+
+  /**
+   * An list of ProductionVariant objects, one for each model that you want to host at this endpoint.
+   * Identifies a model that you want to host and the resources to deploy for hosting it.
+   * If you are deploying multiple models, tell Amazon SageMaker how to distribute traffic among the models by specifying variant weights.
+   */
+  readonly productionVariants: ProductionVariant[];
+
+  /**
+   * Tags to be applied to the endpoint configuration.
+   *
+   * @default - No tags
+   */
+  readonly tags?: { [key: string]: string };
+}
+
+/**
+ * A Step Functions Task to create a SageMaker endpoint configuration
+ *
```

Comment:
include a link to Step Functions API documentation (https://docs.aws.amazon.com/step-functions/latest/dg/connect-sagemaker.html)